test/config_parser_SUITE.erl

```diff
@@ -1337,82 +1336,69 @@ s2s_max_retry_delay(_Config) ->
+        end,
+    ?eqf(modopts(mod_auth_token, [{{validity_period, access}, {13, minutes}},
+                                  {{validity_period, refresh}, {31, days}}]),
+         P([#{<<"token">> => <<"access">>, <<"value">> => 13, <<"unit">> => <<"minutes">>},
+            #{<<"token">> => <<"refresh">>, <<"value">> => 31, <<"unit">> => <<"days">>}])),
+    ?errf(P([#{<<"token">> => <<"access">>, <<"value">> => <<"13">>,
+               <<"unit">> => <<"minutes">>}])),
+    ?errf(P([#{<<"token">> => <<"access">>, <<"value">> => 13, <<"unit">> => <<"minute">>}])),
+    ?errf(P([#{<<"token">> => <<"Access">>, <<"value">> => 13, <<"unit">> => <<"minutes">>}])),
+    ?errf(P([#{<<"value">> => 13, <<"unit">> => <<"minutes">>}])),
+    ?errf(P([#{<<"token">> => <<"access">>, <<"unit">> => <<"minutes">>}])),
+    ?errf(P([#{<<"token">> => <<"access">>, <<"value">> => 13}])).
 
 mod_bosh(_Config) ->
-    run_multi(mod_bosh_cases()).
-
-mod_bosh_cases() ->
     T = fun(K, V) -> #{<<"modules">> => #{<<"mod_bosh">> => #{K => V}}} end,
     M = fun(K, V) -> modopts(mod_bosh, [{K, V}]) end,
-    [?_eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
-     ?_eqf(M(inactivity, infinity), T(<<"inactivity">>, <<"infinity">>)),
-     ?_eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
-     ?_eqf(M(max_wait, infinity), T(<<"max_wait">>, <<"infinity">>)),
-     ?_eqf(M(server_acks, true), T(<<"server_acks">>, true)),
-     ?_eqf(M(server_acks, false), T(<<"server_acks">>, false)),
-     ?errf(T(<<"inactivity">>, -1)),
-     ?errf(T(<<"inactivity">>, <<"10">>)),
-     ?errf(T(<<"inactivity">>, <<"inactivity">>)),
-     ?errf(T(<<"max_wait">>, <<"10">>)),
-     ?errf(T(<<"max_wait">>, -1)),
-     ?errf(T(<<"server_acks">>, -1))].
+    ?eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
+    ?eqf(M(inactivity, infinity), T(<<"inactivity">>, <<"infinity">>)),
+    ?eqf(M(inactivity, 10), T(<<"inactivity">>, 10)),
+    ?eqf(M(max_wait, infinity), T(<<"max_wait">>, <<"infinity">>)),
+    ?eqf(M(server_acks, true), T(<<"server_acks">>, true)),
+    ?eqf(M(server_acks, false), T(<<"server_acks">>, false)),
+    ?eqf(M(maxpause, 10), T(<<"max_pause">>, 10)),
+    ?errf(T(<<"inactivity">>, -1)),
+    ?errf(T(<<"inactivity">>, <<"10">>)),
+    ?errf(T(<<"inactivity">>, <<"inactivity">>)),
+    ?errf(T(<<"max_wait">>, <<"10">>)),
+    ?errf(T(<<"max_wait">>, -1)),
+    ?errf(T(<<"server_acks">>, -1)),
+    ?errf(T(<<"maxpause">>, 0)).
 
 mod_caps(_Config) ->
-    run_multi(mod_caps_cases()).
-
-mod_caps_cases() ->
     T = fun(K, V) -> #{<<"modules">> => #{<<"mod_caps">> => #{K => V}}} end,
     M = fun(K, V) -> modopts(mod_caps, [{K, V}]) end,
-    [?_eqf(M(cache_size, 10), T(<<"cache_size">>, 10)),
-     ?_eqf(M(cache_life_time, 10), T(<<"cache_life_time">>, 10)),
-     ?_errf(T(<<"cache_size">>, -1)),
-     ?_errf(T(<<"cache_size">>, <<"infinity">>)),
-     ?_errf(T(<<"cache_life_time">>, -1)),
-     ?_errf(T(<<"cache_life_time">>, <<"cache_life_time">>))].
```

Comment:
Just one (or two) question from my side:
- you changed `cache_life_time` to `infinity` because the previous one wouldn't make its way to this place in code due to preceding validators, so it hadn't any sense here, correct?
- you changed - in my impression still - correctly erroneous value of `-1` to `0` cause zero is now the closest one value to the boundary value defined per `positive integer`, as `-1` was to `non-negative integer`, correct? I mean `-1` would still fail these tests as well and I'm trying to find the reasoning behind this

Reply:
The reason was indeed to bring the values closer to the boundary, e.g. 0 for the `positive` validator.
We could test both (-1 and 0), but I was trying not to have too much verbosity in tests. There are no strict rules here as long as one successful and one failing case is tested for each option. It's good to have some variety as well, e.g. not to test the same value for all options to have more chance of catching bugs.